README.md

```diff
@@ -8,77 +8,65 @@ npm install hypercore-protocol
 
 [![build status](https://travis-ci.org/mafintosh/hypercore-protocol.svg?branch=master)](https://travis-ci.org/mafintosh/hypercore-protocol)
 
+For detailed info on the messages sent on each channel see [simple-hypercore-protocol](https://github.com/mafintosh/simple-hypercore-protocol)
+
 ## Usage
 
 ``` js
-var protocol = require('hypercore-protocol')
+const Protocol = require('hypercore-protocol')
 
 // create two streams with hypercore protocol
-var streamA = protocol({ id: 'a' })
-var streamB = protocol({ id: 'b' })
+const streamA = new Protocol(true)
+const streamB = new Protocol(false)
 
 // open two feeds specified by a 32 byte key
-var key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
-var feed = streamA.feed(key)
-var remoteFeed = streamB.feed(key)
+const key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
+const feed = streamA.open(key)
+const remoteFeed = streamB.open(key, {
+  // listen for data in remote feed
+  ondata (message) {
+    console.log(message.value.toString())
+  }
+})
 
 // add data to feed
 feed.data({ index: 1, value: '{ block: 42 }'})
 
-// listen data in remoteFeed
-remoteFeed.on('data', function (message) {
-  console.log(message.value.toString())
-})
-
 streamA.pipe(streamB).pipe(streamA)
 ```
 
 `output => { block: 42 }`
 
 ## API
 
-#### `var stream = protocol([options])`
+#### `const stream = new Protocol(initiator, [options])`
 
 Create a new protocol duplex stream.
 
 Options include:
 
 ``` js
 {
-  id: optionalPeerId, // you can use this to detect if you connect to yourself
-  live: keepStreamOpen, // signal to the other peer that you want to keep this stream open forever
-  ack: false, // Explicitly ask a peer to acknowledge each received block
-  userData: opaqueUserData // include user data that you can retrieve on handshake
   encrypt: true, // set to false to disable encryption if you are already piping through a encrypted stream
   timeout: 5000, // stream timeout. set to 0 or false to disable.
-  extensions: [], // names of extensions to use for replication. Must be sorted alphanumerically for handshaking to work
+  keyPair: { publicKey, secretKey }, // use this keypair for the stream authentication
+  onhandshake () { }, // function called when the stream handshake has finished
+  onopen (discoveryKey) { } // function called when the remote stream opens a feed you have not
```

Comment:
These handlers are never invoked as far as i could trace, they're only referenced on line 52 of index.js.
And according to that line, the `onopen` handler is expected to be among the feed-handlers,
while the conditional stream-handler is expected to be named `onremoteopen`.

Reply:
fixed